src/history/cam_hist_file.F90

```diff
@@ -0,0 +1,2093 @@
+            call cam_pio_handle_error(ierr, 'config_define_file: failed to add "units" attribute to "mdt" variable')
+
+            !
+            ! Create variables for model timing and header information
+            !
+            if (split_file_index == instantaneous_file_index) then
+               ierr=pio_def_var (this%hist_files(split_file_index),'ndcur   ',pio_int,(/timdim/),this%ndcurid)
+               call cam_pio_handle_error(ierr, 'config_define_file: failed to define "ndcur" variable')
+               str = 'current day (from base day)'
+               ierr=pio_put_att (this%hist_files(split_file_index), this%ndcurid, 'long_name', trim(str))
+               call cam_pio_handle_error(ierr, 'config_define_file: failed to add "long_name" attribute to "ndcur" variable')
+
+               ierr=pio_def_var (this%hist_files(split_file_index),'nscur   ',pio_int,(/timdim/),this%nscurid)
+               call cam_pio_handle_error(ierr, 'config_define_file: failed to define "nscur" variable')
+               str = 'current seconds of current day'
+               ierr=pio_put_att (this%hist_files(split_file_index), this%nscurid, 'long_name', trim(str))
+               call cam_pio_handle_error(ierr, 'config_define_file: failed to add "long_name" attribute to "nscur" variable')
+               ierr=pio_def_var (this%hist_files(split_file_index),'nsteph',pio_int,(/timdim/),this%nstephid)
+               call cam_pio_handle_error(ierr, 'config_define_file: failed to define "nsteph" variable')
+               str = 'current timestep'
+               ierr=pio_put_att (this%hist_files(split_file_index), this%nstephid, 'long_name', trim(str))
+               call cam_pio_handle_error(ierr, 'config_define_file: failed to add "long_name" attribute to "nsteph" variable')
+            end if
+
+         end if ! .not. satfile
+
+         !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+         !
+         ! Create variables and attributes for field list
+         !
+         !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+         do field_index = 1, size(this%field_list)
+            if (.not. is_satfile .and. .not. restart .and. .not. is_initfile) then
+               if (split_file_index == accumulated_file_index) then
+                  ! this is the accumulated file of a potentially split
+                  ! history tape - skip instantaneous fields
+                  if (this%field_list(field_index)%accumulate_type() == 'lst') then
+                     cycle
+                  end if
+               else
+                  ! this is the instantaneous file of a potentially split
+                  ! history tape - skip accumulated fields
+                  if (this%field_list(field_index)%accumulate_type() /= 'lst') then
+                     cycle
+                  end if
+               end if
+            end if
+
+            if (this%precision() == 'real32') then
+               ncreal = pio_real
+            else
```

Comment:
I wonder if this `else` would be better written out explicitly as `else if(this%precision() == 'REAL64')` ? `hist_type_t`'s `config_precision()` appears to explicitly write out `REAL32`, `REAL64`, and support for a third custom specified option?

Also not sure if this `==` operator is case-insensitive here, it appears that `config_precision()` returns a uppercase string but a comparison is made to lowercase `real32` here.

Reply:
good catch! added `else if` and switched to upper case.